File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.java
import java.util.*;
 
class Solution {
    public int solution(int n) {
        int[] answer = new int[n+1];
      
        
        for(int i = 0; i <= n; ++i) {
            if (i==0){
                answer[i] = 0;
            } else if (i==1){
                answer[i] = 1;
            } else{
                int sum = answer[i-2]+answer[i-1];
                answer[i] = sum % 1234567;
            }
        }
        
        return answer[n];
    }
}

File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.py
def solution(n):
    answer = 0
    n1 = 0
    n2 = 1
    sum = 0
    n3 = 0
    if n == 2:
        answer = sum % 1234567
    for i in range(n-1):
        sum = n1 + n2
        temp = n2
        n2 = n1 + n2
        n1 = temp
    return sum % 1234567